packages/providers/src/provider.ts

```diff
@@ -702,6 +701,7 @@ export default class Provider {
       filter: { owner: owner.toB256(), assetId: assetId && hexlify(assetId) },
     });
 
+    // eslint-disable-next-line @typescript-eslint/no-non-null-assertion
     const coins = result.coins.edges!.map((edge) => edge!.node!);
```

Comment:
```suggestion
    const coins = result.coins.edges.map((edge) => edge.node);
```

Ther `!` assertions are redundant because the underlying types don't have `undefined` or `null` in their type definitions.